### PR TITLE
[WebNN] QDQ's axis should be used for broadcasting

### PR DESCRIPTION
**Pull request by Honry**
For per-axis quantization/dequantization, WebNN requires the scale and zero_point inputs to be broadcastable. Axis should be used for reshape these two inputs.

### CODE REVIEW

**Comment by Honry** [comment]
@fdwr, @guschmue, PTAL, thanks!

File: onnxruntime/core/providers/webnn/builders/impl/qdq_op_builder.cc [hunk 1]
```diff
@@ -49,12 +51,55 @@ Status QDQOpBuilder::AddToModelBuilderImpl(ModelBuilder& model_builder,
...
+    zero_point_shape = target_shape;
+    emscripten::val reshape_scale_options = emscripten::val::object();
+    reshape_scale_options.set("label", node.Name() + "_reshape_scale");
+    scale = model_builder.GetBuilder().call<emscripten::val>("reshape",
+                                                             scale,
+                                                             emscripten::val::array(target_shape),
+                                                             reshape_scale_options);
+
+    if (has_zero_point) {
+      // Insert ones before and after the axis dimension for broadcasting of zero_point tensor.
```
Comment (fdwr) [inline onnxruntime/core/providers/webnn/builders/impl/qdq_op_builder.cc:87]:
```suggestion
      // Reshape the zero_point tensor too.
```
Given the target shape is already calculated above, and no ones are inserted here.

File: onnxruntime/core/providers/webnn/builders/impl/qdq_op_builder.cc [hunk 2]
```diff
@@ -49,12 +51,55 @@ Status QDQOpBuilder::AddToModelBuilderImpl(ModelBuilder& model_builder,
...
+  // For per-axis quantization/dequantization and axis is not equal to input_rank - 1,
+  // we need to reshape the scale and zero_point tensors to make them broadcastable with the input tensor.
+  if (scale_shape.size() == 1 && input_rank > 1 &&
+      block_size == 0 && axis != static_cast<int32_t>(input_rank - 1)) {
+    // Insert ones before and after the axis dimension for broadcasting of scale tensor.
+    std::vector<uint32_t> target_shape{SafeInt<uint32_t>(input_shape[axis])};
+    target_shape.insert(target_shape.begin(), axis, 1);
+    target_shape.insert(target_shape.end(), input_rank - axis - 1, 1);
+    // zero_point has the same shape as the scale tensor.
+    zero_point_shape = target_shape;
```
Comment (fdwr) [inline onnxruntime/core/providers/webnn/builders/impl/qdq_op_builder.cc:78]: std::vector<uint32_t> target_shape(GetTargetShape());

File: onnxruntime/core/providers/webnn/builders/impl/qdq_op_builder.cc (hunk 1 above)
Comment (Honry) [reply onnxruntime/core/providers/webnn/builders/impl/qdq_op_builder.cc]: Fixed, PTAL again. :)

**Review by fdwr — APPROVED** [review]
👍

**Pull request merged**